### PR TITLE
feat(sdk-coin-xlm): root key support for xlm

### PR DESCRIPTION
**Pull request by dwei-bitgo**
TICKET: WP-1418

### CODE REVIEW

File: modules/sdk-coin-xlm/src/lib/keyPair.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (this.keyPair.prv) {
+      const privateKey = stellar.StrKey.encodeEd25519SecretSeed(Buffer.from(this.keyPair.prv, 'hex'));
+      result.prv = raw ? stellar.StrKey.decodeEd25519SecretSeed(privateKey).toString('hex') : privateKey;
+    }
+
+    return result;
+  }
+
+  /** @inheritdoc */
+  getAddress(): string {
```
Comment (dwei-bitgo) [inline modules/sdk-coin-xlm/src/lib/keyPair.ts:29]: Not sure if this is necessary for XLM

Comment (alebusse) [reply modules/sdk-coin-xlm/src/lib/keyPair.ts:29]: I think its just the pub key in XLM format, so just return the pub

File: modules/sdk-coin-xlm/src/xlm.ts [hunk 2]
```diff
@@ -246,7 +259,7 @@ export class Xlm extends BaseCoin {
    * @returns is it valid?
    */
   isValidPub(pub: string): boolean {
-    return stellar.StrKey.isValidEd25519PublicKey(pub);
+    return stellar.StrKey.isValidEd25519PublicKey(pub) || isValidEd25519PublicKey(pub);
```
Comment (dpkjnr) [inline modules/sdk-coin-xlm/src/xlm.ts:262]: why both? it looks confusing

Comment (dwei-bitgo) [reply modules/sdk-coin-xlm/src/xlm.ts:262]: the stellar validation method only permits the stellar specific G-prefix version of public key, it fails to validate raw root keys in hex form, that is what the second method is there for, will add a code comment to clarify

**Review by alebusse — APPROVED** [review]

**Review by CapnMigraine — APPROVED** [review]

**Pull request merged**